### PR TITLE
Implemented new resource manager as per feedback

### LINKED ISSUE #4214: New Resource Manager observations
Hi,
I like the new resource manager module but I've found a couple things that I think would improve a lot the experience.

1. As far as I can see there's only a single view with the big icons. I'd suggest adding an alternate view showing all details (name, date created, date modified, size, ...). And then allow users to alternate between list view and icon view

2. There's a drop down with values Item name, Last modified on date, created on date. 
I guess this is the sorting of items but notice it does not say it anywhere. 
Suggestion would be to change the text to "Order by..." 
It would also be nice to have an ASC/DESC option
It would also be nice to have a sort by size option

3. User of popups not consistent
I see a few options that open a popup, for example when trying to delete an item.
Any other option seems to open a new section at the end of the listing. I find it quite confusing, even more if the list of items in the folder is large. It's not obvious that you click on edit penzil for example and then you need to scroll down to see the details of the item.
My suggestion would be to open all those "editor" sections in a popup.

Also notice that if you click the edit penzil on the first item of a two row list of items, the second rows gets moved.
This would be fixed if we used popups instead, so not a big deal if you choose to change to popups.

imagen

4. Multiple seleccion
This is related to number 1 above.
There's no option to select multiple items at once. So if you want to move, delete, download, ... you need do to that one by one.
I'd suggest a selectbox on the list view to allow for this option.

5. Edit and move should close previous edit sections
Not an issue if we used popups as discussed in number 3
Take a file, and click the penzil icon to edit. Then click the icon to move.
You end up with two sections option at the bottom of the page.
Very confusing

6. Permissions for files
When editing a folder I see the option to manage permissions.
That doesn't seem to be available when editing a file.

7. 
Create a folder TEST and add a file to it.
Click edit iconon the file. Without closing the edit section below click the parent folder on the breadcrumb
Click move icon on the folder.
You end up with two open sections at the botton: one to edit the previous file, one to move the folder.
Again, this all would be solved if using modal popups for editors.

8. Spacing
Some spacing required here
imagen

### LINKED ISSUE #4338: File suffix with Resource Manager in 9.8.0
I am populating a brand new install of 9.8.0 with the Telerik components removed.

I needed to upload a number of images from my client of various sorts (png and jpg). THe filenames had extensions mostly in lower case but 3 images were labelled with .JPG. The Resource Manager refused to upload these.

I went to the allowed file types and added JPG as one of these and retried - again the file were rejected.

Finally I change the filenames extensions to lowercase jpg and the files were accepted and uploaded with no problem.

I am attaching two versions of one of these files in both variants to see if others have this issue.
SettingOut3
SettingOut3

### LINKED ISSUE #4611: Resource Manager Enhancement - Add Alternate List View
_Note that this is 1 of 3, split from the original Issue #4390 after the suggestion by @valadas._

## Description of problem
The current view of the folders and files in the new RM of DNN 9.08.00 (thru 9.09.01 so far) is very difficult to visually navigate, especially when there are large numbers of files and folders.

## Description of solution
Provide an option to switch to/set a more readable and compact list view, with full filenames visible including the extension. 

## Description of alternatives considered
For "round one" I would like to encourage this feature to happen sooner and use a rudimentary or simple implementation. There are a lot of industry standard examples out there. Using the old file manager as a basic starting point would be reasonable too? Just listing the files and folders in rows with a few basic click actions would help. Ideally, the list view could also include the "date last modified" column.

## Screenshots

Screenshot 1 - Resource Manager (current view)
image

Screenshot 2 - Existing Dnn File Manager 
image
  
## Additional context
A use case where the current "Tile View" is a serious hinderance is when you have a large site with a lot of content. Just a few dozen rows of giant Tiles becomes difficult to navigate, especially with the currently truncated filenames.

## Affected browsers
* [x] All

### LINKED ISSUE #4612: Resource Manager Enhancement - Improve Card/Tile View
_Note that this is 2 of 3, split from the original Issue #4390 after the suggestion by @valadas._

## Description of problem
The current view of the folders and files in the Resource Manager that became available as an optional install as of DNN 9.08.00 is difficult to visually navigate and use manage files and folders, especially once more than a dozen folders or files or both is in view. 
 
The actions (Edit, Move, etc) are hidden (and not even hover-labeled, see #4384) and even relatively short file names are unnecessarily even more shortened, unless you hover on each one. Plus, every card is very bulky, so even a folder with a dozen files or little more becomes unnecessarily "long," requiring a lot of scrolling. Files and folders have cards with the same background (and poor contrast with the text), and the image previews are mostly hidden (see screenshot 1).

## Description of solution
Enhance the existing Tiled View to provide more info using a more modern UX/UI; Already visible actions+labels, where the file/folder names are shortened only above 60-100 characters (or so) and the file type always visible.

Other options and ideas to consider (see Screenshot 2, below as a visual rough draft of some of these ideas):

- improving contrast and information hierarchy (when several folders or non-image files of the same type, their name is more important than the icon)
- increasing file name limit before shortening (the font could be also smaller, once contrast is improved)
- making extensions always visible for files
- adding "order by" file type
- making the action icons always visible; they could be moved on top instead of on the right side to make space for the filename and just slightly reduced in size; obviously add labels/tooltips to the actions, as suggested in #4398 
- widening the clipping of the file preview/thumbnail, where applicable 
- adding a selection checkbox to each card for bulk actions (move, etc)
- taking more advantage of responsive styling and flexbox capabilities instead of absolute positioning and fixed widths/heights for the cards and their contents

## Screenshots

Screenshot 1 of Resource Manager (current view as of Dnn 9.08.x thru 9.09.01 so far)
image

Screenshot 2 of an Enhanced View as re-imagined by @Valeria-B (see #4390)
image

## Affected browsers
* [x] All

### LINKED ISSUE #5227: Resource Manager Installer Deploys Un-Needed Files
## Description of bug
The Resource Manager module when installed includes three web.config files within the working directory of the solution..

* Web.config
* Web.Debug.Config
* Web.Release.Config

These files are un-necessary on a production environment, at a minimum the second two should not be included in the sites.

## Steps to reproduce
List the precise steps to reproduce the bug:
1. Install Resource Manager
2. Look at the /DesktopModules/ResourceManager folder

## Current behavior
Files are included

## Expected behavior
Files should not be.

## Affected version

* [ ] 10.00.00 alpha build
* [ X ] 09.10.02 release candidate
* [ ] 09.10.01 latest supported release

### PR DESCRIPTION
**Pull request by valadas**
This new version of the resource manager uses WebComponents. It implements the requested UI/UX improvements requested by many. It no longer depends on the React common components.

If you want to test this, you can download the build result from DevOps and do a clean install.
I would also love some testing on upgrades from 9.10.2 to 9.11.0.
Would love for people using Group/User mode to give this a test too.

Resolves #4214 
Resolves #4338 
Resolves #4611 
Resolves #4612 
Resolves #5227

### CODE REVIEW

File: DNN Platform/Modules/ResourceManager/ResourceManager.Web/package.json [hunk 1]
```diff
@@ -1,72 +1,36 @@
 {
-  "name": "resource-library",
-  "version": "9.11.0",
-  "description": "Resource Library",
-  "private": true,
+  "name": "dnn-resource-manager",
+  "version": "9.10.2",
+  "description": "Stencil Component Starter",
```
Comment (mitchelsellers) [inline DNN Platform/Modules/ResourceManager/ResourceManager.Web/package.json:4]: Lets rename this to reference the DNN Resource Manager product name

File: DNN Platform/Modules/ResourceManager/ResourceManager.Web/readme.md [hunk 2]
```diff
@@ -0,0 +1,13 @@
...
+
+# dnn-resource-manager Web Component
+
+This component is made to interact with a DNN backend.
+
+# Stencil
```
Comment (mitchelsellers) [inline DNN Platform/Modules/ResourceManager/ResourceManager.Web/readme.md:7]: We need to make sure that this is properly referenced in the license file as well

File: DNN Platform/Modules/ResourceManager/ResourceManager.dnn [hunk 3]
```diff
@@ -108,5 +108,35 @@
...
+      <owner>
+        <name>.NET Foundation and Contributors</name>
+        <organization>DNN Community</organization>
+        <url>https://dnncommunity.org</url>
+        <email>info@dnncommunity.org</email>
+      </owner>
+      <license src="License.txt"></license>
+      <releaseNotes src="ReleaseNotes.txt"></releaseNotes>
+      <dependencies>
+        <dependency type="CoreVersion">09.02.00</dependency>
```
Comment (mitchelsellers) [inline DNN Platform/Modules/ResourceManager/ResourceManager.dnn:123]: Is this correct?

**Review by mitchelsellers — CHANGES_REQUESTED** [review]

File: DNN Platform/Modules/ResourceManager/ResourceManager.Web/package.json (hunk 1 above)
Comment (david-poindexter) [inline DNN Platform/Modules/ResourceManager/ResourceManager.Web/package.json:4]:
```suggestion
  "description": "Resource Manager",
```

File: DNN Platform/Modules/ResourceManager/ResourceManager.dnn [hunk 4]
```diff
@@ -108,5 +108,35 @@
         </component>
       </components>
     </package>
+    <package name="PersonaBar.ResourceManager" type="PersonaBar" version="09.10.02">
```
Comment (valadas) [inline DNN Platform/Modules/ResourceManager/ResourceManager.dnn:111]:
```suggestion
    <package name="PersonaBar.ResourceManager" type="PersonaBar" version="09.11.00">
```

File: DNN Platform/Modules/ResourceManager/ResourceManager.dnn (hunk 3 above)
Comment (valadas) [inline DNN Platform/Modules/ResourceManager/ResourceManager.dnn:123]:
```suggestion
        <dependency type="CoreVersion">09.11.00</dependency>
```

**Review by valadas — COMMENTED** [review]
A few things in the manifest

File: SolutionInfo.cs [hunk 5]
```diff
@@ -13,6 +13,6 @@
 [assembly: AssemblyProduct("https://dnncommunity.org")]
 [assembly: AssemblyCopyright("DNN Platform is copyright 2002-2020 by .NET Foundation. All Rights Reserved.")]
 [assembly: AssemblyTrademark("DNN")]
-[assembly: AssemblyVersion("9.11.0.0")]
-[assembly: AssemblyFileVersion("9.11.0.0")]
-[assembly: AssemblyInformationalVersion("9.11.0.0 Custom build")]
+[assembly: AssemblyVersion("9.10.2")]
+[assembly: AssemblyFileVersion("9.10.2.0")]
+[assembly: AssemblyInformationalVersion("9.10.2 Custom build")]
```
Comment (valadas) [inline SolutionInfo.cs:18]:
```suggestion
[assembly: AssemblyVersion("9.11.0.0")]
[assembly: AssemblyFileVersion("9.11.0.0")]
[assembly: AssemblyInformationalVersion("9.11.0.0 Custom build")]

```

**Review by mitchelsellers — APPROVED** [review]
THANK YOU!!!

**Comment by valadas** [comment]
> THANK YOU!!!

**Comment by valadas** [comment]
I'll take a better look at the build issues tomorrow (well today)...

**Review by donker — APPROVED** [review]
Awesome!

**Pull request merged**